backend/models/user.js

```diff
@@ -0,0 +1,85 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+
+const STATES = [ 'AK', 'AL', 'AR', 'AS', 'AZ', 'CA', 'CO', 'CT', 'DC', 'DE', 'FL', 'GA', 'GU', 'HI', 'IA', 'ID', 'IL', 'IN', 'KS', 'KY', 'LA', 'MA', 'MD', 'ME', 'MI', 'MN', 'MO', 'MS', 'MT', 'NC', 'ND', 'NE', 'NH', 'NJ', 'NM', 'NV', 'NY', 'OH', 'OK', 'OR', 'PA', 'PR', 'RI', 'SC', 'SD', 'TN', 'TX', 'UT', 'VA', 'VI', 'VT', 'WA', 'WI', 'WV', 'WY'];
+
+const validateEmail = email => {
+  var re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  return re.test(String(email).toLowerCase());
+}
+
+
+const userSchema = new Schema({
+  name: {
+    first: {
+      type: String,
+      required: [true, 'Please enter first name.'],
+      trim: true,
+      maxlength: 50
+    },
+    last: {
+      type: String,
+      required: [true, 'Please enter last name.'],
+      trim: true,
+      maxlength: 50
+    }
+  },
+  username: {
+    type: String,
+    required: [true, 'Please enter username.'],
+    unique: true,
+    lowercase: true,
+    trim: true,
+    minlength: [4, 'Please select username 4 characters or more.'],
+    maxlength: [20, 'Please select username 20 characters or less.'],
```

Comment:
Maybe use 'Please enter a username of 4 characters or more' and 'Please enter a username of 20 characters or less'. Not a big deal, but I feel that it reads better this way.